genai-perf/genai_perf/profile_data_parser/llm_profile_data_parser.py

```diff
@@ -77,9 +80,29 @@ def __init__(
         goodput_constraints: Dict[str, float] = {},
     ) -> None:
         self._tokenizer = tokenizer
+        self._session_metrics: SessionMetrics = defaultdict(lambda: defaultdict(list))
         super().__init__(filename, goodput_constraints)
 
-    def _parse_requests(self, requests: dict) -> Metrics:
+    def _parse_profile_data(self, data: dict) -> None:
+        """Parse through the entire profile data to collect statistics."""
+        self._profile_results = {}
+        for experiment in data["experiments"]:
+            infer_mode = experiment["experiment"]["mode"]
+            load_level = experiment["experiment"]["value"]
+            requests = experiment["requests"]
+
+            llm_metrics = self._parse_requests(requests)
+
+            # aggregate and calculate statistics
+            statistics = Statistics(llm_metrics)
+            self._profile_results[(infer_mode, str(load_level))] = statistics
+
+            # calculate per-session statistics
+            for session_id, session_metric in self._session_metrics.items():
```

Comment:
Will this still work without sessions?

Reply:
Yes, it will just skip since `self._session_metrics` is empty.